azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java

```diff
@@ -33,11 +42,12 @@
      * Must be configured when OAuth2 authentication is done in front end
      */
     private String clientSecret;
+
     /**
      * Azure AD groups.
      */
-    @NotEmpty
```

Comment:
This loosens the validation. I was unsure if adding a new custom validator for the class to check either this or the new property was populated after construction.

Reply:
I think the validation put here is for ensuring at least one ad group should be available at runtime. The property without this may makes the configuration from required to optional here.

Reply:
On that understanding I can add a custom validator to cover or validation?

Reply:
@Incarnation-p-lee validator added to check either property has a value.